tools/config/__init__.py

```diff
@@ -69,7 +69,7 @@
 
 
 ALLOWED_FEATURES = [
-    "BOOTLOADER", "BLE", "LWIP", "STORAGE", "NANOSTACK", "CRYPTOCELL310",
+    "BOOTLOADER", "BLE", "LWIP", "STORAGE", "NANOSTACK", "CRYPTOCELL310", "PSA", "TFM", "TFM_TWINCPU", "TFM_V8M", "MBED_PSA_SRV", "PSA_SRV_IPC", "PSA_SRV_IMPL", "PSA_SRV_EMUL", "NSPE",
```

Comment:
Why isn't Mbed TLS listed as an allowed feature here? It's present in `features/mbedtls`. If this list isn't for enumerating all possible features, what's this list for?

Reply:
@Patater I'm not sure why `features/mbedtls` is not listed here. @mark-edgeworth any thoughts.
if `features_add` config option is mentioned in `targets.json` then https://github.com/ARMmbed/mbed-os/blob/master/tools/config/__init__.py#L1347 is invoked which checks the added feature against `ALLOWED_FEATURES`.

Reply:
Sorry, no. This is not something that we'd be involved with.

Reply:
If we change this, this doc should be updated : https://github.com/ARMmbed/mbed-os-5-docs/blob/46e78f5b7b8c38c383820af3adc9a85d758c0c75/docs/reference/configuration/mbed_targets.md#features-features_add-and-features_remove

I dont think this change makes sense, so many features ? They do not look to me like a complete feature (for instance if I use TFM why do I need to add also PSA and the rest) ?

Reply:
This list looks outdated though, lwip and storage and nanostack are not features anymore. same as psa/tfm should not be?

Reply:
> They do not look to me like a complete feature

So should these be components? There are so many of them so that we can control the globbing build system well enough. We should have some documentation about how these filters all work (and would need to update it as well, along with all other Mbed OS + PSA Docs), although I can't find the component explaining docs right now. @devran01 Do we have a link we could share with @0xc0170 ?

Reply:
Thanks for the summary @Patater . @0xc0170 Unfortunately, these are not complete features but something that we need to support single v7-M, dual v7-M and v8-M targets. 
The document describing the component structure is https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/porting/psa/spm.md#defining-a-psa-compliant-target-in-targetsjson.
Some more details https://github.com/ARMmbed/mbed-os/pull/11640#issuecomment-540997520.

I think it's better to update `mbed-os-docs` once we merge this feature branch onto master. But before that we need to agree on adding PSA as a feature and accept having incomplete features combined together or revert back to components as before.

Reply:
not components. But as there are so many. Why are these not coming from targets itself (part of config) ? There are like 8 of them added here. How can I combine them (is it even possible to just use one or two?) ?

Why would I care about any of these to be added as "feature" ?:
""PSA", "TFM", "TFM_TWINCPU", "TFM_V8M", "MBED_PSA_SRV", "PSA_SRV_IPC", "PSA_SRV_IMPL", "PSA_SRV_EMUL", "NSPE""

most of them have TFM or PSA prefix, Why if I want PSA these are not coming in as well? Or TFM? I do not understand how they are related so take my comment as a outsider :)